server/middleware.go

```diff
@@ -0,0 +1,35 @@
+package server
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/flashbots/go-template/util"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+func (s *Server) logger(next http.Handler) http.Handler {
```

Comment:
I'd say remove the middleware example, and make sure the zap middleware from https://github.com/flashbots/go-utils/blob/main/httplogger/httplogger.go#L111 is working correctly?

Probably replace sugared logger by regular logger there